packages/cli/src/lib/bundler/server.ts

```diff
@@ -60,6 +60,9 @@ export async function serveBundle(options: ServeOptions) {
         // Paths with dots should still use the history fallback.
         // See https://github.com/facebookincubator/create-react-app/issues/387.
         disableDotRule: true,
+
+        // The index needs to be rewritten relative to the new public path, including subroutes.
+        index: `${config.output?.publicPath}index.html`,
```

Comment:
What does this fix? Serving on a sub-route already works, but is there something extra that this does?

Reply:
The existing functionality doesn't handle re-GETs, reloading the page on http://localhost:3000/test returns a 404 since it's trying to get http://localhost:3000/index.html not http://localhost:3000/test/index.html.